src/plugins/packages/stacked-barchart/src/StackedBarchartPanel.tsx

```diff
@@ -0,0 +1,274 @@
+import React, { useLayoutEffect, useMemo } from 'react';
+import ReactDOM from 'react-dom';
+import useMeasure from 'react-use-measure';
+import { PanelProps, classicColors } from '@grafana/data';
+import { useTheme } from '@grafana/ui';
+import { css } from '@emotion/css';
+import * as d3 from 'd3';
+
+import { BarDatum, StackedBarchartOptions } from './types';
+import { prepareGraphableFields, countDigits } from './utils';
+
+interface Props extends PanelProps<StackedBarchartOptions> {}
+
+export const StackedBarchartPanel: React.FC<Props> = ({ options, data, width, height }) => {
+  const theme = useTheme();
+  const graphRef = React.useRef(null);
+  const graphTooltipRef = React.useRef(null);
+  const [legendRef, legendMeasure] = useMeasure();
+  const legendHeight = options.showLegend ? legendMeasure.height : 0;
+  const svgHeight = height - legendHeight;
+
+  const { stacks, xDistinctGroups, yDistinctGroups, xMaxGroupCount, error } = useMemo(
+    () => prepareGraphableFields(data?.series, options),
+    [data, options]
+  );
+
+  const margin = {
+    top: 5,
+    right: 5,
+    bottom: 40,
+    left: 20 + 12 * countDigits(xMaxGroupCount),
+  };
+  const colorScale = d3.scaleOrdinal().domain(yDistinctGroups).range(classicColors);
+
+  useLayoutEffect(() => {
+    if (error) {
+      return;
+    }
+    const tooltip = d3.select(graphTooltipRef.current).style('display', 'none');
```

Comment:
Nice catch) I haven't noticed this because `d3` seems to conveniently return an empty selection when the input is `null`) Added the check for null values)